examples/reporting/parallel_report_download.rb

```diff
@@ -0,0 +1,186 @@
+  # omitted here.
+  threads = []
+  query_list.each do |query_key, query|
+    customer_ids.each do |cid|
+      cid = cid.tr("-", "")
+      # Starts the report download in a background thread.
+      threads << Thread.new do
+        begin
+          puts "Requesting #{query_key} for CID #{cid}"
+          responses = ga_service.search_stream(
+            customer_id: cid.tr("-", ""),
+            query: query,
+          )
+          # Stores the number of rows for each report for illustrative purposes.
+          # Users of this code example can implement other logic here such as
+          # storing response to a database/file, pass the response on to
+          # another method for further processing, etc.
+          num_rows = 0
+          responses.each do |response|
+            response.results.each do |row|
+              num_rows += 1
+            end
+          end
+          reports_succeeded << {
+            :cid => cid,
+            :query_key => query_key,
+            :num_rows => num_rows,
+          }
+        rescue Google::Ads::GoogleAds::Errors::GoogleAdsError => e
+          error_messages = ""
+          e.failure.errors.each do |error|
+            error_messages += error.message
+          end
+          reports_failed << {
+            :cid => cid,
+            :query_key => query_key,
+            :error_messages => error_messages,
+          }
+        end
+      end
+    end
+  end
+
+  puts "Awaiting results from #{threads.size} report download requests..."
+
+  # Waits for all pending requests to the current set of customer IDs
+  # to complete.
+  threads.each { |aThread|  aThread.join }
+
+  puts 'Download completed, results:'
+  puts 'Successful reports:'
+  while !reports_succeeded.empty? do
+    result = reports_succeeded.pop()
+    puts "Customer ID: #{result[:cid]}, Query Key: #{result[:query_key]}, " \
+      "Total rows retrieved: #{result[:num_rows]}"
+  end
+  puts 'Failed reports:'
+  while !reports_failed.empty? do
+    result = reports_failed.pop()
+    puts "Customer ID: #{result[:cid]}, Query Key: #{result[:query_key]}, " \
+      "Error Messages: #{result[:error_messages]}"
+  end
+  puts 'End of results.'
+end
+
+if __FILE__ == $PROGRAM_NAME
+  PAGE_SIZE = 1000
+
+  options = {}
+
+  # The following parameter(s) should be provided to run the example. You can
+  # either specify these by changing the INSERT_XXX_ID_HERE values below, or on
+  # the command line.
+  #
+  # Parameters passed on the command line will override any parameters set in
+  # code.
+  #
+  # Running the example with -h will print the command line usage.
+  options[:customer_ids] = [
+    'INSERT_CUSTOMER_ID_1_HERE',
+    'INSERT_CUSTOMER_ID_2_HERE',
+  ]
+
+  OptionParser.new do |opts|
+    opts.banner = sprintf('Usage: ruby %s [options]', File.basename(__FILE__))
+
+    opts.separator ''
+    opts.separator 'Options:'
+
+    opts.on('-C', '--customer-ids CUSTOMER-IDS', String, 'Customer ID') do |v|
```

Comment:
Add a little more information to the help text explaining how/why multiple customer IDs are implemented.

Reply:
Added some more help text to explain, for customer ID and login customer ID as well.